Various fixes, use upload by date-time and added a limit to auto-tagger.

ItsKaa opened the pull request:
Couple of improvements and some fixes for crashes that I found.

* Handling various exceptions:
  * pixiv_tags was defined inside a try-except but was also used outside of it causing in a consistent crash of a url I cannot remember.
  * unhandled extensions: handled the general error under `not post_id` because the return value was incorrect and also excluded some extra extensions.
  * JSONDecodeError, I believe the `name` couldn't be found or some other element, best to not crash because of it.
* Fixed tag sanitization not always being consistent.
* Added a limit to the auto-tagger, there's an underlying problem where it seems to take up too many file handles, I haven't really find the cause and it only happens when dealing with a lot of files, a limit of 1000 works fine for me. Also might help some people out dealing with API limits.
* Uploading files based on the file date, either from the metadata or the modified date (for accurate dates, it will require setting this up in the gallery-dl config file). I enjoy having things uploaded in order, especially when doing a bulk import.

reluce reviewed (changes requested):
In `szuru_toolkit.py` I'm replacing whitespaces with underscores, because the user might input their tag string with whitespaces included, like: `--add-tags "foo, bar"`.

So it would tag the post with ` bar` when `bar` is intended. Replacing it with `_` would end up with `_bar`.

Sanitizing the tags in `auto_tagger.py` makes sense though, I don't know why I wasn't `sanitize_tags` even used anymore in the project.

It would probably make sense to drop  `.replace(' ', '').split(',')` everywhere in `szuru_toolkit.py` and use `sanitize_tags()` in each of the sub scripts like `tag_posts.py`.

ItsKaa commented:
> In `szuru_toolkit.py` I'm replacing whitespaces with underscores, because the user might input their tag string with whitespaces included, like: `--add-tags "foo, bar"`.
> 
> So it would tag the post with ` bar` when `bar` is intended. Replacing it with `_` would end up with `_bar`.
> 
> Sanitizing the tags in `auto_tagger.py` makes sense though, I don't know why I wasn't using `sanitize_tags()` anymore in the project.
> 
> It would probably make sense to drop `.replace(' ', '').split(',')` everywhere in `szuru_toolkit.py` and use `sanitize_tags()` in each of the sub scripts like `tag_posts.py`.

That's fair, didn't think of that scenario. Was just thinking of replacing it with
```
list(x.strip().replace(' ', '_') for x in add_tags.split(','))
```
Which works but  I saw that there are a total of 10x `replace(' ', '')` calls, and it's probably better to refactor those at some point and try to get everything to use `sanatize_tags` anyway.

So for now I just reverted it.

reluce commented:
Can you run pre-commit to run a few checks and possibly fix them as well? Then I'll go ahead and merge, thanks!

ItsKaa commented:
> Can you run pre-commit to run a few checks and possibly fix them as well? Then I'll go ahead and merge, thanks!

Done!
